### DISCUSSION TITLE
openresty and ngx.print

### DISCUSSION THREAD

**Discussion by mnegovanovic** [minoki/LunarML#15] [Q&A] [answered]
Hi,

I am playing around with LunarML and OpenResty. How do i do simple `ngx.print` from SML? I have this code but it compiles ngx.print as string value:

```
structure export = struct                                                            
  fun dispatch () =                                                                  
      let                                                              
          val ngx_print = Lua.unsafeToValue "ngx.print"     
      in
          Lua.call1 ngx_print #[Lua.fromString "Hello world from LunarML!"]                                                         
      end                                                                                                                           
  end
```

It compiles ... but like i said `ngx.print` is pulled in as string value. Do i need to add new primitive op?

Regards
Milos

**Comment by minoki** [comment, marked as answer]
Use `Lua.global` to access the global variable `ngx`, and `Lua.field` to access its field.

So, something like `val ngx_print = Lua.field (Lua.global "ngx", "print")` should work.
